lib/pacemaker/pcmk_rule.c

```diff
@@ -155,58 +155,58 @@ crm_rule_check(pcmk__output_t *out, pe_working_set_t *data_set,
      */
     xpath = crm_strdup_printf("//rule[@id='%s']", rule_id);
     xpath_obj = xpath_search(cib_constraints, xpath);
-    max = numXpathResults(xpath_obj);
+    num_results = numXpathResults(xpath_obj);
 
-    if (max == 0) {
-        rc = ENXIO;
+    free(xpath);
+    freeXpathObject(xpath_obj);
+
+    if (num_results == 0) {
         out->err(out, "No rule found with ID=%s", rule_id);
-        goto done;
-    } else if (max > 1) {
-        rc = ENXIO;
+        return ENXIO;
+    } else if (num_results > 1) {
         out->err(out, "More than one rule with ID=%s found", rule_id);
-        goto done;
+        return ENXIO;
     }
 
-    free(xpath);
-    freeXpathObject(xpath_obj);
-
     /* Next, make sure it has exactly one date_expression. */
     xpath = crm_strdup_printf("//rule[@id='%s']//date_expression", rule_id);
     xpath_obj = xpath_search(cib_constraints, xpath);
-    max = numXpathResults(xpath_obj);
+    num_results = numXpathResults(xpath_obj);
 
-    if (max != 1) {
-        rc = EOPNOTSUPP;
+    free(xpath);
+    freeXpathObject(xpath_obj);
+
+    if (num_results != 1) {
         out->err(out, "Can't check rule %s because it does not have exactly "
                  "one date_expression", rule_id);
-        goto done;
+        return EOPNOTSUPP;
     }
 
-    free(xpath);
-    freeXpathObject(xpath_obj);
-
     /* Then, check that it's something we actually support. */
     xpath = crm_strdup_printf("//rule[@id='%s']//date_expression["
                               "@operation!='date_spec']", rule_id);
     xpath_obj = xpath_search(cib_constraints, xpath);
-    max = numXpathResults(xpath_obj);
+    num_results = numXpathResults(xpath_obj);
 
-    if (max == 0) {
-        free(xpath);
+    free(xpath);
+
+    if (num_results == 0) {
         freeXpathObject(xpath_obj);
 
         xpath = crm_strdup_printf("//rule[@id='%s']//date_expression["
                                   "@operation='date_spec' "
                                   "and date_spec/@years "
                                   "and not(date_spec/@moon)]", rule_id);
         xpath_obj = xpath_search(cib_constraints, xpath);
-        max = numXpathResults(xpath_obj);
+        num_results = numXpathResults(xpath_obj);
 
-        if (max == 0) {
-            rc = ENXIO;
+        free(xpath);
+
+        if (num_results == 0) {
             out->err(out, "Rule must either not use date_spec, or use "
                      "date_spec with years= but not moon=");
-            goto done;
+            freeXpathObject(xpath_obj);
+            return ENXIO;
```

Comment:
Seems like `EOPNOTSUPP` might be better here too

Reply:
Done; will revert if we're concerned about anyone depending on `ENXIO` specifically in this situation